packages/backend/src/app-connection/app-connection-service.ts

```diff
@@ -0,0 +1,162 @@
+import { apId, AppConnection, AppConnectionId, AppConnectionType, BaseOAuth2ConnectionValue, CloudOAuth2ConnectionValue, Cursor, OAuth2ConnectionValueWithApp, ProjectId, RefreshTokenFromCloudRequest, SeekPage, UpsertConnectionRequest } from "shared";
+import { databaseConnection } from "../database/database-connection";
+import { buildPaginator } from "../helper/pagination/build-paginator";
+import { paginationHelper } from "../helper/pagination/pagination-utils";
+import { AppConnectionEntity } from "./app-connection-entity";
+import axios, { AxiosError } from "axios";
+import qs from "qs";
+
+const appConnectionRepo = databaseConnection.getRepository(AppConnectionEntity);
+
+export const appConnectionService = {
+    async upsert(request: UpsertConnectionRequest): Promise<AppConnection> {
+        await appConnectionRepo.upsert({ ...request, id: apId() }, ["name", "projectId"]);
```

Comment:
FYI, upsert docs says: `Unlike save method executes a primitive operation without cascades, relations and other operations included.`